### PR TITLE
[POAE7-1462] Use Velox round function to replace Omnisci built-in round

### PR DESCRIPTION
**Pull request by xieqi**

### CODE REVIEW

**Comment by winningsix** [comment]
@Intel-bigdata/bdf-cider-engine

File: QueryEngine/RuntimeFunctions.cpp [hunk 1]
```diff
@@ -258,6 +258,7 @@ DEF_UMINUS_NULLABLE(double, double)
       const from_type operand,                                                 \
       const from_type from_null_val,                                           \
       const to_type to_null_val) {                                             \
+    fprintf(stderr, "call round\n");                                           \
```
Comment (jikunshang) [inline QueryEngine/RuntimeFunctions.cpp:261]: could remove this?

**Review by jikunshang — APPROVED** [review]

**Comment by winningsix** [comment]
Will this involve loop dependency here? Velox -> Omnisci -> Velox

Also, we need to make up the missing pieces how Spark/Prestodb functions are registered in Omnisic as well as how functions are called via ```RelAlgExecutionUnit```. Any thoughts for this part?

File: QueryEngine/ExtensionFunctions.hpp [hunk 2]
```diff
@@ -185,6 +187,17 @@ double radians(const double x) {
 
 EXTENSION_NOINLINE
 double Round(const double x, const int32_t y) {
+  fprintf(stderr, "omnisci call velox Round\n");
+  using funcClass = typename facebook::velox::functions::udf_round<double>::template udf<facebook::velox::core::DynamicExec>;
```
Comment (winningsix) [inline QueryEngine/ExtensionFunctions.hpp:191]: There're two different implements for Spark and presto in Velox. How can we handle this?
One in prestosql\RegisterArithmetic.cpp and one in sparksql/RegisterArithmetic.cpp

**Review by winningsix — CHANGES_REQUESTED** [review]
Two follow ups:
1. Rebase this
2. Handle different implementations in Presto and Spark.

**Comment by yma11** [comment]
LGTM. Just remember refine the function related contents in plan doc. Thanks.

**Review by winningsix — APPROVED** [review]
LGTM. Need a rebase before merge.

**Pull request merged**